web/src/components/FileViewerModal/FileViewerModal.provider.tsx

```diff
@@ -41,28 +42,40 @@ const FileViewerModalProvider = ({children}: IProps) => {
     type: 'definition',
   });
   const [getJUnit] = useLazyGetJUnitByRunIdQuery();
-  const [getTestDefinitionYaml] = useLazyGetTestDefinitionYamlByRunIdQuery();
+  const [getResourceDefinition] = useLazyGetResourceDefinitionQuery();
+  const [fileProps, setProps] = useState({
+    title: '',
+    language: '',
+    subtitle: '',
+    fileName: '',
+  });
 
   const loadJUnit = useCallback(
     async (testId: string, runId: string) => {
       const data = await getJUnit({runId, testId}).unwrap();
       setIsFileViewerOpen(true);
       setFileViewerData({data, type: 'junit'});
+      setProps(propsMap.junit);
     },
     [getJUnit]
   );
 
-  const loadTestDefinitionYaml = useCallback(
-    async (testId: string, version: number) => {
-      const data = await getTestDefinitionYaml({testId, version}).unwrap();
+  const loadDefinition = useCallback(
```

Comment:
This provider now handles the definition fetch and display for tests, transactions and environments